newton/_src/sensors/sensor_contact.py

```
@@ -273,6 +258,11 @@ def eval(self, contacts: Contacts):
         Args:
             contacts (Contacts): The contact data to evaluate.
         """
+        if contacts.force is None:
+            raise ValueError(
+                "SensorContact requires a ``Contacts`` object with ``force`` allocated."
+                "Create ``SensorContact`` before ``Contacts`` for automatically requesting it."
+            )
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)